packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityPlugin.java

```diff
@@ -62,7 +65,28 @@ public void onCancel(Object arguments) {
     receiver = null;
   }
 
-  private static String getNetworkType(int type) {
+  private String getNetworkType(int type, ConnectivityManager manager) {
+    if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      Network network = manager.getActiveNetwork();
+      NetworkCapabilities capabilities = manager.getNetworkCapabilities(network);
+      if (capabilities == null) {
+        return "none";
+      }
+      if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)
+          || capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
+        return "wifi";
+      }
+      if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
+        return "mobile";
+      }
+    }
+
+    return getNetworkTypeLegacy(type);
+  }
+
+  @SuppressWarnings("deprecated")
+  private String getNetworkTypeLegacy(int type) {
```

Comment:
Maybe `getNetworkTypeLegacy(ConnectivityManager manager)` since 
`NetworkInfo info = manager.getActiveNetworkInfo();` is also deprecated.

Reply:
I see, that makes sense.

Reply:
@cyanglaz I have pushed the changes have a look